### PR TITLE
Move all core types into namespaces

### PR DESCRIPTION
**Pull request by JesseTG**
This PR moves all types in the core to the `melonDS` namespace, to reduce the risk of clashes in the global namespace in melonDS itself and any frontend that uses it as a dependency. It was originally part of [this PR](https://github.com/melonDS-emu/melonDS/pull/1883), but it got too unwieldy. So I'm splitting up into smaller PRs.

This PR looks enormous, but most of the changes are one-line `using` or `namespace` declarations.

### CODE REVIEW

File: src/ARM.cpp [hunk 1]
```diff
@@ -104,7 +106,7 @@ const u32 ARM::ConditionTable[16] =
     0x0000  // NE
 };
 
-ARM::ARM(u32 num, ARMJIT::ARMJIT& jit, Melon::GPU& gpu) :
+ARM::ARM(u32 num, ARMJIT& jit, melonDS::GPU& gpu) :
```
Comment (RSDuck) [inline src/ARM.cpp:109]: why do you prefix the GPU?

**Review by RSDuck — COMMENTED** [review]
I'm not happy with how the melonDS namespace creeps into the vendored dependencies, though I don't really know how to make it better either.

File: src/ARM.cpp (hunk 1 above)
Comment (JesseTG) [reply src/ARM.cpp:109]: Because having a member with the same name as a type causes C++ to choke, and naming the type as `class GPU` looks weird.

**Comment by JesseTG** [comment]
> I'm not happy with how the melonDS namespace creeps into the vendored dependencies, though I don't really know how to make it better either.

This was primarily to accommodate melonDS's integer `typedef`s. If you'd like, I can replace them (in the dependencies) with standard equivalents.

**Pull request merged**